static/src/javascripts/projects/common/modules/onward/breaking-news.js

```diff
@@ -154,64 +159,72 @@ const renderSpectre = ($breakingNews: bonzo): bonzo =>
-            });
-        }, alertDelay);
-
-        mediator.emit('modules:onwards:breaking-news:ready', true);
-    } else {
-        mediator.emit('modules:onwards:breaking-news:ready', false);
-    }
-    return alert;
+const show = (): void => {
+    const $body = bonzo(document.body);
+    const $breakingNews = bonzo(qwery('.js-breaking-news-placeholder'));
+
+    // if its the first time we've seen this alert, we wait 3 secs to show it
+    // otherwise we show it immediately
+    const alertDelay = knownAlertIDs.hasOwnProperty(alertToShow.id)
+        ? 0
+        : DEFAULT_DELAY;
+
+    // $breakingNews is hidden, so this won't trigger layout etc
+    $breakingNews.append(renderAlert(alertToShow));
+
+    // copy of breaking news banner (with blank content) used inline at the
+    // bottom of the body, so the bottom of the body can visibly scroll
+    // past the pinned alert
+    const $spectre = renderSpectre($breakingNews);
+
+    // inject the alerts into DOM
+    setTimeout(() => {
+        fastdom.write(() => {
+            if (alertDelay === 0) {
+                $breakingNews.removeClass('breaking-news--fade-in');
+            }
+            $body.append($spectre);
+            $breakingNews.removeClass('breaking-news--hidden');
+            markAlertAsSeen(alertToShow.id);
+        });
+    }, alertDelay);
 };
 
-const breakingNewsInit = (): Promise<?Alert> => {
-    if (userCanDismissAlerts()) {
-        knownAlertIDs = local.get(knownAlertIDsStorageKey) || {};
-
-        return fetchBreakingNews()
-            .then(parseResponse)
-            .then(getRelevantAlerts)
-            .then(pruneKnownAlertIDs)
-            .then(filterAlertsByDismissed)
-            .then(filterAlertsByAge)
-            .then(pickNewest)
-            .then(showAlert)
-            .catch(ex => {
-                reportError(ex, {
-                    feature: 'breaking-news',
+const canShow = (): Promise<boolean> =>
+    new Promise(resolve => {
+        if (userCanDismissAlerts()) {
+            knownAlertIDs = local.get(knownAlertIDsStorageKey) || {};
+
+            fetchBreakingNews()
+                .then(parseResponse)
+                .then(getRelevantAlerts)
+                .then(pruneKnownAlertIDs)
+                .then(filterAlertsByDismissed)
+                .then(filterAlertsByAge)
+                .then(pickNewest)
+                .then(alert => {
+                    if (alert) {
+                        alertToShow = alert;
+                        resolve(true);
+                    } else {
+                        resolve(false);
+                    }
+                })
+                .catch(ex => {
+                    reportError(ex, {
+                        feature: 'breaking-news',
+                    });
+                    resolve(false);
                 });
-            });
-    }
-    return Promise.reject(new Error('cannot dismiss'));
+        } else {
+            resolve(false);
```

Comment:
Could return `Promise.resolve(false)` here, and within the `Promise` chain above return `true` or `false` as necessary. Then we don't need the `new Promise(...)` wrapper